packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx

```diff
@@ -318,29 +317,40 @@ export const TearsheetShell = React.forwardRef(
     }
 
     useEffect(() => {
-      if (open && position === depth) {
-        // Focusing the first element or selectorPrimaryFocus element
-        claimFocus(firstElement, modalRef, selectorPrimaryFocus);
+      if (open) {
+        claimFocus();
       }
-    }, [
-      currentStep,
-      depth,
-      firstElement,
-      modalRef,
-      modalRefValue,
-      open,
-      position,
-      selectorPrimaryFocus,
-      hasError,
-    ]);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [open, currentStep, effectiveHasCloseIcon]);
 
     useEffect(() => {
-      if (prevOpen && !open && launcherButtonRef) {
+      if (prevOpen && !open && launcherButtonRef?.current) {
         setTimeout(() => {
-          launcherButtonRef.current.focus();
+          launcherButtonRef?.current.focus();
         }, 0);
       }
-    }, [launcherButtonRef, open, prevOpen]);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [open, prevOpen]);
+
+    useEffect(() => {
+      requestAnimationFrame(() => {
+        if (
+          open &&
+          depth === position &&
+          !modalRef?.current?.contains(document.activeElement)
+        ) {
+          claimFocus();
+        }
+      });
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [depth, position]);
+
+    useEffect(() => {
+      if (hasError && !modalRef?.current?.contains(document.activeElement)) {
+        claimFocus();
```

Comment:
This may go beyond the scope of this review... but should claimFocus be wrapped in a useCallback? That would allow us to add that in the dep array of these useEffects and not worry about runaway re-renders... and we could lift some of these disable next line rules.